pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go

```diff
@@ -69,7 +69,9 @@ type OperatingSystemConfigList struct {
 type OperatingSystemConfigSpec struct {
 	// DefaultSpec is a structure containing common fields used by all extension resources.
 	DefaultSpec `json:",inline"`
-
+	// EnableContainerD indicate if kubelet should be started with ContainerD and not Docker.
+	// +optional
+	EnableContainerD *bool  `json:"enableContainerD,omitempty"`
```

Comment:
This is leftover?

Reply:
No, this is the indication in the OSC for the enablement of ContainerD.
OS extensions using this.

Reply:
Why it would be needed? The OSC will already contain `containerd.service` unit. Isn't this enough for the extension to handle it?

Reply:
In my opinion the explicit enablement of ContainerD is more readable and understood,
The ContainerD unit doesn't even contain a start command but just Drop In unit for an Pre execution init script.
More than that, ContainerD is enabled by default in most OS, Docker already using it even without the enablement of ContainerD in Gardener shoot.

Reply:
I get your point @nimrodoron, but for the sake of of consistency I would probably also use the CRI name here. Concretely, maybe a `CRI struct` with (today) just one field `Name` is better from an API pov. Assuming we will add a third option next to docker and containerd tomorrow, we would have hard-coded containerd in this API which is not too nice.

Reply:
Ok i added CRIConfig.